### PR TITLE
[WIP] Draft of Marc's proposed NVT API revisions

### PR DESCRIPTION
**Pull request by karlhigley**

### CODE REVIEW

**Comment by marcromeyn** [comment]
I think these 2 branches could be a better starting point:

- #990 
- #991

File: nvtabular/workflow.py [hunk 1]
```diff
@@ -178,10 +190,61 @@ def fit_transform(self, dataset: Dataset) -> Dataset:
...
+            DatasetCollection(data=dataset).to_parquet(self.work_dir, overwrite=overwrite, **kwargs)
+
+        return transformed
+
+    def generate_schema(self, dataset, output_path=None):
+        return Schema.calculate_on_dataset(
+            dataset, self.column_group.tags_by_column(), output_path=output_path
+        )
+
+    def fit_transform_collection(
```
Comment (karlhigley) [inline nvtabular/workflow.py:213]: I wonder if we could make `fit_transform` accept and process a variable number of `Datasets` and bypass the need for a second method here?

**Review by karlhigley — COMMENTED** [review]
Looks like there's a lot of good stuff in here! Would love to break this down into smaller pieces we can add incrementally.

**Comment by karlhigley** [comment]
Partially(?) superseded by #990 and #991

**Comment by karlhigley** [comment]
Now broken down into smaller PRs, so closing. Have migrated relevant comments to #990

**Pull request closed without merging**